plugins/woocommerce/tests/php/includes/class-wc-brands-test.php

```diff
@@ -74,6 +82,105 @@ public function test_product_brand_list_shortcode_with_show_empty_brands_arg() {
 		$this->assertStringContainsString( 'Empty Brand', $output );
 	}
 
+	/**
+	 * Test that brand counts are correctly calculated and cached.
+	 */
+	public function test_brand_count_calculation_and_caching() {
+		$data = $this->setup_brand_test_data();
+
+		// Get the brand term.
+		clean_term_cache( $data['brand_with_products']['term_id'], 'product_brand' );
+		$brand_term = $this->get_first_brand_term( array( $data['brand_with_products']['term_id'] ) );
+
+		// Test that the count is correctly calculated.
+		$this->assertEquals( 1, $brand_term->count, 'Brand should have 1 product' );
+
+		// Test that the count is cached in term meta.
+		$cached_count = get_term_meta( $brand_term->term_id, 'product_count_product_brand', true );
+		$this->assertEquals( '1', $cached_count, 'Brand count should be cached in term meta' );
+	}
+
+	/**
+	 * Test that brand counts respect product visibility settings.
+	 */
+	public function test_brand_count_respects_product_visibility() {
+		$data    = $this->setup_brand_test_data();
+		$product = $data['product'];
+
+		// Enable hide out of stock setting FIRST.
+		update_option( 'woocommerce_hide_out_of_stock_items', 'yes' );
+
+		// THEN set product to out of stock (hook will fire with correct setting).
+		$product->set_stock_status( 'outofstock' );
+		$product->save();
+
+		// Get the brand term.
+		$brand_term = $this->get_first_brand_term( array( $data['brand_with_products']['term_id'] ) );
+
+		// Test that the count is 0 when product is out of stock and hidden.
+		$this->assertEquals( 0, $brand_term->count, 'Brand count should be 0 when product is out of stock and hidden' );
```

Comment:
Just as a further clarification about the comments: look at this comment. It literally says exactly what the assertion string below says. As a human, I'm naturally reading the comment as I think it's going to add something valuable, and I'll fill up my “human context” 🙃